pkg/ipamd/ipamd.go

```diff
@@ -1319,13 +1365,7 @@ func (c *IPAMContext) nodeIPPoolReconcile(ctx context.Context, interval time.Dur
 		}
 
 		if c.enablePodENI && metadataResult.TrunkENI != "" {
-			// Label the node that we have a trunk
-			err = c.SetNodeLabel(ctx, "vpc.amazonaws.com/has-trunk-attached", "true")
-			if err != nil {
-				podENIErrInc("askForTrunkENIIfNeeded")
-				log.Errorf("Failed to set node label for trunk. Aborting reconcile", err)
-				return
-			}
+			log.Debugf("Trunk interface (%s) has been added to the node already.", metadataResult.TrunkENI)
```

Comment:
Are we making the assumption here that the event would've been already generated for this node, since the trunk is already present?

Reply:
Yes, since this is in reconciling, we won't send event if the trunk has been attached to the node.

Reply:
We do check `c.dataStore.GetTrunkENI() == "" || !hasTrunkLabel` to send an event if a node has no trunk or has no label.